examples/optimization/demo_3D_wavelets.py

```diff
@@ -0,0 +1,245 @@
+"""
+3D wavelet denoising
+====================================================================================================
+
+This example shows how to use a 3D wavelet denoiser for denoising a 3D image. We first apply a standard soft-thresholding
+wavelet denoiser to a 3D brain MRI volume. We then extend the denoiser objective to a redundant dictionary of wavelet
+bases, which does not admit a closed-form solution. We solve the denoising problem using the Dykstra-like algorithm.
```

Comment:
The resulting algorithm is readily available in :meth:`deepinv.models.WaveletDictDenoiser` class